app/controllers/items/detail.js

```diff
@@ -570,9 +574,28 @@ export default GoodcityController.extend(
       },
 
       async updatePackageType() {
-        const pkgType = await this.get(
-          "packageTypeService"
-        ).userPickPackageType();
+        let pkgType;
+
+        if (this.get("model.isPartOfSet")) {
+          const packageSet = this.get("model.packageSet");
+          const pkg = await this.get("packageService").userPickPackage({
+            packageTypes: this.get("packageService").allChildPackageTypes(
+              packageSet
+            ),
+            storageTypeName: "Package" // we don't add boxes to sets
+          });
```

Comment:
Why is it picking a package instead of a type ? 🤔 
The goal is to select a new type for the existing package, right ?

Reply:
I would imagine something like this :

```javascript
const types = this.get('packageTypeService');
const allowedPackageTypes = this.get("model.isPartOfSet") ?
             types.chidlrenTypesOf(this.get('model.packageSet.packageType')) : 
             types.getAllTypes();

const newType = await this.get("packageTypeService").userPickPackageType({ 
     subsetPackageTypes: allowedPackageTypes
});

// save the new package type
```